core/templates/components/forms/schema-based-editors/schema-based-custom-editor.component.spec.ts

```diff
@@ -93,4 +120,23 @@ describe('Schema Based Custom Editor Component', () => {
 
     expect(component.localValue).toBe('true');
   });
+
+  it(
+    'should register validator and call it when the form validation changes',
+    fakeAsync(() => {
+      component.schema = { obj_type: 'UnicodeString', type: 'custom' };
+      fixture.detectChanges();
+      flush();
+      const spy = jasmine.createSpy('validator onchange spy');
+      let mockEmitter = new EventEmitter();
+      component.registerOnValidatorChange(spy);
+      spyOnProperty(
+        component.hybridForm, 'statusChanges').and.returnValue(mockEmitter);
```

Comment:
Break up the setup and the test.

Try to use better names for "spy" etc. -- I can't really tell what's going on here. Try to give more context using variable names and comments.

Reply:
Done